spec/compiler/semantic/private_spec.cr

```diff
@@ -318,76 +318,64 @@ describe "Semantic: private" do
       )) { int32 }
   end
 
-  it "doesn't find private class from outside namespace" do
-    assert_error %(
-      class Foo
-        private class Bar
-        end
-      end
-
-      Foo::Bar
-      ),
-      "private constant Foo::Bar referenced"
-  end
-
-  it "doesn't find private module from outside namespace" do
-    assert_error %(
-      class Foo
-        private module Bar
-        end
-      end
-
-      Foo::Bar
-      ),
-      "private constant Foo::Bar referenced"
-  end
-
-  it "doesn't find private enum from outside namespace" do
-    assert_error %(
-      class Foo
-        private enum Bar
-          A
-        end
-      end
-
-      Foo::Bar
-      ),
-      "private constant Foo::Bar referenced"
-  end
-
-  it "doesn't find private alias from outside namespace" do
-    assert_error %(
-      class Foo
-        private alias Bar = Int32
-      end
-
-      Foo::Bar
-      ),
-      "private constant Foo::Bar referenced"
-  end
-
-  it "doesn't find private lib from outside namespace" do
-    assert_error %(
-      class Foo
-        private lib LibBar
+  {% for kind, decl in {
+                         "class"    => %(class Bar; end),
+                         "module"   => %(module Bar; end),
+                         "enum"     => %(enum Bar; A; end),
+                         "alias"    => %(alias Bar = Int32),
+                         "lib"      => %(lib Bar; end),
+                         "constant" => %(Bar = 1),
+                       } %}
```

Comment:
These could be reused.

```suggestion
  {%
    definitions = {
      "class"    => %(class Bar; end),
      "module"   => %(module Bar; end),
      "enum"     => %(enum Bar; A; end),
      "alias"    => %(alias Bar = Int32),
      "lib"      => %(lib Bar; end),
      "constant" => %(Bar = 1),
    }
  %}
  {% for kind, decl in definitions %}
```

Reply:
But the definitions are not identical! This would need more transformation at which point it's unreasonable to reuse.

Reply:
You're right.